dask/dataframe/io/tests/test_parquet.py

```diff
@@ -4557,6 +4557,47 @@ def test_pyarrow_filesystem_option(tmp_path, fs):
     assert_eq(ddf, df)
 
 
+@PYARROW_MARK
+def test_fsspec_to_parquet_filesystem_option(tmp_path):
+    from fsspec import get_filesystem_class
+    from pyarrow import dataset as pa_da
+
+    key1 = "/read1"
+    key2 = str(tmp_path / "write1")
+
+    df = pd.DataFrame({"a": range(10)})
+    fs = get_filesystem_class("memory")(use_instance_cache=False)
+    pa_da.write_dataset(
+        pa.Table.from_pandas(df),
+        key1,
+        format="parquet",
+        filesystem=fs,
+    )
+
+    # read in prepared data
+    ddf = dd.read_parquet(
+        key1,
+        engine="pyarrow",
+        filesystem=fs,
+    )
+    ddf.to_parquet(key2, engine="pyarrow", filesystem=fs)
+
+    # make sure we didn't write to local fs
+    assert len(list(tmp_path.iterdir())) == 0, "wrote to local fs"
+
+    # make sure we wrote a key to memory fs
+    assert len(fs.ls(key2, detail=False)) == 1
+
+    # ensure append functionality works
+    ddf.to_parquet(key2, engine="pyarrow", append=True, filesystem=fs)
+    assert len(fs.ls(key2, detail=False)) == 2, "should have two parts"
+
+    rddf = dd.read_parquet(key2, engine="fastparquet", filesystem=fs)
```

Comment:
```suggestion
    rddf = dd.read_parquet(key2, engine="pyarrow", filesystem=fs)
```
There is no guarantee that `fastparquet` is installed when this test runs.

Reply:
Ah makes sense.